cookbook/controller/upload_file.rst

```diff
@@ -164,15 +187,203 @@ There are some important things to consider in the code of the above controller:
    that information. That's why it's always better to generate a unique name and
    use the :method:`Symfony\\Component\\HttpFoundation\\File\\UploadedFile::guessExtension`
    method to let Symfony guess the right extension according to the file MIME type;
-#. The ``UploadedFile`` class also provides a :method:`Symfony\\Component\\HttpFoundation\\File\\UploadedFile::move`
-   method to store the file in its intended directory. Defining this directory
-   path as an application configuration option is considered a good practice that
-   simplifies the code: ``$this->container->getParameter('brochures_dir')``.
 
-You can now use the following code to link to the PDF brochure of an product:
+You can use the following code to link to the PDF brochure of an product:
+
+.. configuration-block::
+
+    .. code-block:: html+twig
+
+        <a href="{{ asset('uploads/brochures/' ~ product.brochure) }}">View brochure (PDF)</a>
+
+    .. code-block:: html+php
+
+        <a href="<?php echo $view['assets']->getUrl('uploads/brochures/'.$product->getBrochure()) ?>">
+            View brochure (PDF)
+        </a>
+
+Creating an Uploader Service
+----------------------------
+
+To avoid logic in controllers, making them big, you can extract the upload
```

Comment:
Not sure if I really like this service, it's small doesn't do much and doesn't seem to make things simpler in any way. However, it felt like a nice step between a big controller and using a Doctrine listener.